apiserver/facades/agent/caasagent/caasagent.go

```diff
@@ -4,14 +4,173 @@
 package caasagent
 
 import (
+	"context"
+
+	"github.com/juju/names/v6"
+
 	"github.com/juju/juju/apiserver/common"
-	"github.com/juju/juju/apiserver/common/cloudspec"
-	"github.com/juju/juju/apiserver/common/model"
+	commonmodel "github.com/juju/juju/apiserver/common/model"
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/internal"
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/core/crossmodel"
+	coremodel "github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/network"
+	corewatcher "github.com/juju/juju/core/watcher"
+	environscloudspec "github.com/juju/juju/environs/cloudspec"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/state"
 )
 
-// FacadeV2 is the V2 facade of the caas agent
+// StubService will be replaced once the implementation is finished.
+type StubService interface {
+	// CloudSpec returns the cloud spec for the model.
+	CloudSpec(ctx context.Context) (environscloudspec.CloudSpec, error)
+}
+
+// ModelService represents the credential service provided by the
```

Comment:
This sentence computes like 1/0 in my head. What about:

`ModelService provide a means for access common information related to the current model`